### ISSUE TITLE
`list activations` is empty when it should not be.

### ISSUE THREAD

**Issue by JiniousChoi** [apache/incubator-openwhisk-deploy-kube#323]
```
$ wsk -i list | grep greeting 
/guest/greeting                                                        private nodejs:6

$ wsk -i action invoke -r greeting 
{
    "msg": "Hello, stranger from somewhere!"
}

$ wsk -i activation list
activations
```

And waiting for it to appear for several minutes doesn't make any difference.
Help me please.

**Comment by JiniousChoi** [comment]
Just in case:
```
$ helm status owdev | grep '==> v1/Pod(related)' -A 13
==> v1/Pod(related)

NAME                         READY  STATUS     RESTARTS  AGE
invoker-f8fvd                1/1    Running    0         4h
invoker-trv6l                1/1    Running    0         4h
apigateway-7b87dd957f-w7c95  1/1    Running    0         4h
couchdb-6b87ccfb78-tpd6r     1/1    Running    0         4h
nginx-598c75d7d6-5shbr       1/1    Running    0         4h
redis-5d77674f65-kqtnc       1/1    Running    0         4h
controller-0                 1/1    Running    1         4h
kafka-0                      1/1    Running    0         4h
zookeeper-0                  1/1    Running    0         4h
init-couchdb-w4ttr           0/1    Completed  0         4h
```

**Comment by dgrove-oss** [comment]
It's not unusual to have to do `wsk -i activations list` a couple of times in a row to see the most recent activations.  They are being stored in CouchDB which is eventually consistent.   If listing several times in a row doesn't show the new activations, then there might be something wrong.  If so, please post the logs from the invoker and the couchdb pods.

**Comment by JiniousChoi** [comment]
I know what you mean. I have openwhisk installed on my private cloud.
Sometimes, activations doesn't appear right away but they became consistent eventually.
But this time, it's been almost a day and I still cannot  come up with any of the expected activations.

I am totally new to kubenetes, so let me get back to you after some learning.
I might take a day or so, so that I can debug and/or share the issue by the pods' logs.

**Comment by JiniousChoi** [comment]
When I invoke `wsk -i activation list`, the only log from couchdb pod is this:
```
[notice] 2018-10-25T07:32:32.712897Z couchdb@couchdb0 <0.28926.8> e3f9ab89a0 couchdb.openwhisk.svc.cluster.local:5984 10.192.0.3 whisk_admin GET /test_activations/_design/whisks.v2.1.0/_view/activations?descending=true&endkey=%5B%22guest%22,0%5D&reduce=false&startkey=%5B%22guest%22,%22%EF%BF%B0%22,%22%EF%BF%B0%22%5D&limit=30&stale=update_after 200 ok 2
```

**Comment by JiniousChoi** [comment]
And am I supposed to have such dbs in couchdb?
```
root@couchdb-6b87ccfb78-tpd6r:~# curl http://localhost:5984/_all_dbs
["_global_changes","_replicator","_users","ow_kube_couchdb_initialized_marker","test_activations","test_subjects","test_whisks"]
```

**Comment by dgrove-oss** [comment]
Sorry for the slow response; I was on vacation much of last week.

The list of database in couchdb looks right; it exactly matches what I see in my deployed system.  (The three `_` databases are internal implementation dbs used by couchdb; the `ow_kube_couchdb_initialized_marker` means that the `init-couchdb` Job completed successfully.  The three `test_` databases are the "real" databases actually being used by OpenWhisk.

The controller output corresponding to `wsk -i activation list` also looks as expected.  

So my current best guess is that something went wrong when the invoker attempted to put the activations for the executed actions into the database.  It might be worth looking at your invoker log to see if there is anything reported there around the action invocation.  I will capture some logs locally and put into a separate comment to show what they should look like.

**Comment by dgrove-oss** [comment]
In a fresh deployment of the Helm chart, I did the following `wsk` operations
```
daves-mbp:incubator-openwhisk-deploy-kube dgrove$ wski activation list
activations
daves-mbp:incubator-openwhisk-deploy-kube dgrove$ wski action create hello ../myActions/hello.js 
ok: created action hello
daves-mbp:incubator-openwhisk-deploy-kube dgrove$ wski action invoke -r hello
{
    "body": "Hello World"
}
daves-mbp:incubator-openwhisk-deploy-kube dgrove$ wski activation list
activations
daves-mbp:incubator-openwhisk-deploy-kube dgrove$ wski activation list
activations
58fc694788dc4c83bc694788dc2c8344 hello               
```

Here is the controller log:
```
[2018-10-29T17:56:50.816Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] GET /api/v1/namespaces/_/activations limit=30&skip=0
[2018-10-29T17:56:50.816Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] [BasicAuthenticationDirective] authenticate: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502
[2018-10-29T17:56:50.817Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] [Identity] [GET] serving from cache: CacheKey(23bc46b1-71f6-4ed5-8c54-816aa4f8c502) [marker:database_cacheHit_count:2]
[2018-10-29T17:56:50.818Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] [CouchDbRestStore] [QUERY] 'test_activations' searching 'whisks.v2.1.0/activations [marker:database_queryView_start:3]
[2018-10-29T17:56:50.824Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] [CouchDbRestStore]  [marker:database_queryView_finish:9:6]
[2018-10-29T17:56:50.825Z] [INFO] [#tid_vbqIZeyOYKlewMYZ4yvjOC0EFwEFC3s8] [BasicHttpService] [marker:http_get.200_count:10:10]
[2018-10-29T17:56:58.320Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] PUT /api/v1/namespaces/_/actions/hello overwrite=false
[2018-10-29T17:56:58.321Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [BasicAuthenticationDirective] authenticate: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502
[2018-10-29T17:56:58.322Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [Identity] [GET] serving from cache: CacheKey(23bc46b1-71f6-4ed5-8c54-816aa4f8c502) [marker:database_cacheHit_count:1]
[2018-10-29T17:56:58.326Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [WhiskAction] [GET] serving from datastore: CacheKey(guest/hello) [marker:database_cacheMiss_count:6]
[2018-10-29T17:56:58.326Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [CouchDbRestStore] [GET] 'test_whisks' finding document: 'id: guest/hello' [marker:database_getDocument_start:6]
[2018-10-29T17:56:58.337Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [CouchDbRestStore]  [marker:database_getDocument_finish:17:11]
[2018-10-29T17:56:58.337Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [WhiskAction] invalidating CacheKey(guest/hello)
[2018-10-29T17:56:58.338Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [WhiskAction] write initiated on new cache entry
[2018-10-29T17:56:58.338Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [CouchDbRestStore] [PUT] 'test_whisks' saving document: 'id: guest/hello, rev: null' [marker:database_saveDocument_start:18]
[2018-10-29T17:56:58.351Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [CouchDbRestStore]  [marker:database_saveDocument_finish:31:13]
[2018-10-29T17:56:58.352Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [WhiskAction] write all done, caching CacheKey(guest/hello) Cached
[2018-10-29T17:56:58.352Z] [INFO] [#tid_rkS6xTJxBv0DRgxdLrC32SGezb8S8LDm] [BasicHttpService] [marker:http_put.200_count:32:32]
[2018-10-29T17:57:07.328Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] POST /api/v1/namespaces/_/actions/hello blocking=true&result=true
[2018-10-29T17:57:07.329Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [BasicAuthenticationDirective] authenticate: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502
[2018-10-29T17:57:07.329Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [Identity] [GET] serving from cache: CacheKey(23bc46b1-71f6-4ed5-8c54-816aa4f8c502) [marker:database_cacheHit_count:1]
[2018-10-29T17:57:07.375Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [WhiskActionMetaData] [GET] serving from datastore: CacheKey(guest/hello) [marker:database_cacheMiss_count:47]
[2018-10-29T17:57:07.375Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore] [GET] 'test_whisks' finding document: 'id: guest/hello' [marker:database_getDocument_start:47]
[2018-10-29T17:57:07.382Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore]  [marker:database_getDocument_finish:54:7]
[2018-10-29T17:57:07.416Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ActionsApi]  [marker:controller_blockingActivation_start:88]
[2018-10-29T17:57:07.417Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ActionsApi] action activation id: 58fc694788dc4c83bc694788dc2c8344 [marker:controller_loadbalancer_start:88]
[2018-10-29T17:57:07.423Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ShardingContainerPoolBalancer] posting to 'invoker0/docker-for-desktop' with activation id '58fc694788dc4c83bc694788dc2c8344' [marker:controller_kafka_start:95]
[2018-10-29T17:57:07.428Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ShardingContainerPoolBalancer] posted to invoker0[0][2] [marker:controller_kafka_finish:100:4]
[2018-10-29T17:57:07.429Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ActionsApi]  [marker:controller_loadbalancer_finish:101:12]
[2018-10-29T17:57:07.639Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ShardingContainerPoolBalancer] received active ack for '58fc694788dc4c83bc694788dc2c8344'
[2018-10-29T17:57:07.640Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ActionsApi]  [marker:controller_blockingActivation_finish:312:224]
[2018-10-29T17:57:07.642Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [BasicHttpService] [marker:http_post.200_count:314:314]
[2018-10-29T17:57:10.529Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] GET /api/v1/namespaces/_/activations limit=30&skip=0
[2018-10-29T17:57:10.530Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] [BasicAuthenticationDirective] authenticate: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502
[2018-10-29T17:57:10.530Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] [Identity] [GET] serving from cache: CacheKey(23bc46b1-71f6-4ed5-8c54-816aa4f8c502) [marker:database_cacheHit_count:2]
[2018-10-29T17:57:10.532Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] [CouchDbRestStore] [QUERY] 'test_activations' searching 'whisks.v2.1.0/activations [marker:database_queryView_start:3]
[2018-10-29T17:57:10.541Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] [CouchDbRestStore]  [marker:database_queryView_finish:13:10]
[2018-10-29T17:57:10.542Z] [INFO] [#tid_LObjC1RpmuE5C8qsZnsxjvQUKcaMdIzr] [BasicHttpService] [marker:http_get.200_count:14:14]
[2018-10-29T17:57:11.564Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] GET /api/v1/namespaces/_/activations limit=30&skip=0
[2018-10-29T17:57:11.564Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] [BasicAuthenticationDirective] authenticate: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502
[2018-10-29T17:57:11.565Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] [Identity] [GET] serving from cache: CacheKey(23bc46b1-71f6-4ed5-8c54-816aa4f8c502) [marker:database_cacheHit_count:1]
[2018-10-29T17:57:11.566Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] [CouchDbRestStore] [QUERY] 'test_activations' searching 'whisks.v2.1.0/activations [marker:database_queryView_start:2]
[2018-10-29T17:57:11.574Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] [CouchDbRestStore]  [marker:database_queryView_finish:10:8]
[2018-10-29T17:57:11.575Z] [INFO] [#tid_jfX1dxAOGp14Yrd24jv0A18HGL1TLisP] [BasicHttpService] [marker:http_get.200_count:11:11]
```

Here is the invoker log
```
[2018-10-29T17:57:07.432Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [InvokerReactive]  [marker:invoker_activation_start:104]
[2018-10-29T17:57:07.500Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [WhiskAction] [GET] serving from datastore: CacheKey(guest/hello) [marker:database_cacheMiss_count:172]
[2018-10-29T17:57:07.500Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore] [GET] 'test_whisks' finding document: 'id: guest/hello, rev: 1-b4f39c1aa00716db5039c5e0babbb4c5' [marker:database_getDocument_start:172]
[2018-10-29T17:57:07.541Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore]  [marker:database_getDocument_finish:213:41]
[2018-10-29T17:57:07.544Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ContainerPool] containerStart containerState: prewarmed action: hello namespace: guest activationId: 58fc694788dc4c83bc694788dc2c8344 [marker:invoker_containerStart.prewarmed_count:216]
[2018-10-29T17:57:07.544Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [DockerContainer] sending initialization to ContainerId(a3389c766331b495d2d36e248f63b5a40af756b26babcc2cd7f6f62c0e5e658e) ContainerAddress(172.17.0.3,8080) [marker:invoker_activationInit_start:216]
[2018-10-29T17:57:07.546Z] [INFO] [#tid_sid_invokerWarmup] [DockerClientWithFileAccess] running /usr/bin/docker run -d --cpu-shares 256 --memory 256m --memory-swap 256m --network bridge -e __OW_API_HOST=https://192.168.65.3:31001 --name wskdocker-for-desktop0_4_prewarm_nodejs6 --cap-drop NET_RAW --cap-drop NET_ADMIN --ulimit nofile=1024:1024 --pids-limit 1024 --log-driver json-file openwhisk/nodejs6action:latest (timeout: 1 minute) [marker:invoker_docker.run_start:6421498]
[2018-10-29T17:57:07.619Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [DockerContainer] initialization result: ok [marker:invoker_activationInit_finish:291:75]
[2018-10-29T17:57:07.619Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [DockerContainer] sending arguments to /guest/hello at ContainerId(a3389c766331b495d2d36e248f63b5a40af756b26babcc2cd7f6f62c0e5e658e) ContainerAddress(172.17.0.3,8080) [marker:invoker_activationRun_start:291]
[2018-10-29T17:57:07.630Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [DockerContainer] running result: ok [marker:invoker_activationRun_finish:301:10]
[2018-10-29T17:57:07.630Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ContainerProxy]  [marker:invoker_collectLogs_start:302]
[2018-10-29T17:57:07.636Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [InvokerReactive] posted completion of activation 58fc694788dc4c83bc694788dc2c8344
[2018-10-29T17:57:07.647Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [ContainerProxy]  [marker:invoker_collectLogs_finish:319:17]
[2018-10-29T17:57:07.648Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore] [PUT] 'test_activations' saving document: 'id: guest/58fc694788dc4c83bc694788dc2c8344, rev: null' [marker:database_saveDocument_start:320]
[2018-10-29T17:57:07.649Z] [INFO] [#tid_sid_dbBatcher] [CouchDbRestStore] 'test_activations' saving 1 documents [marker:database_saveDocumentBulk_start:6421601]
[2018-10-29T17:57:07.662Z] [INFO] [#tid_sid_dbBatcher] [CouchDbRestStore]  [marker:database_saveDocumentBulk_finish:6421614:13]
[2018-10-29T17:57:07.663Z] [INFO] [#tid_0uyY46AvCWe6GFktvoBqCvJT7BLmXzrX] [CouchDbRestStore]  [marker:database_saveDocument_finish:334:14]
[2018-10-29T17:57:07.718Z] [INFO] [#tid_sid_invokerNanny] [DockerClientWithFileAccess] running /usr/bin/docker pause a3389c766331b495d2d36e248f63b5a40af756b26babcc2cd7f6f62c0e5e658e (timeout: 10 seconds) [marker:invoker_docker.pause_start:6421670]
[2018-10-29T17:57:07.748Z] [INFO] [#tid_sid_invokerNanny] [DockerClientWithFileAccess]  [marker:invoker_docker.pause_finish:6421698:28]
[2018-10-29T17:57:08.428Z] [INFO] [#tid_sid_invokerWarmup] [DockerClientWithFileAccess]  [marker:invoker_docker.run_finish:6422380:882]
```
If you can get similar logs, look in the invoker log and verify that logs are being collected and there is a successful write to the test_activations document after the execution of the action completes.

**Comment by dgrove-oss** [comment]
I fixed a  bug specific to running on kubeadm-dind-cluster and invoker health last night (#329).  I haven't fully updated the documentation yet.  Make sure you have this commit, and then try adding the following to your `myCluster.yaml` and redeploying the helm chart.
```yaml
invoker:
  containerFactory:
    dind: true
```

**Comment by JiniousChoi** [comment]
Thanks. I'll give it a try.
By the way, to make sure, I am not so sure with the following quote from the main README.md on this project:

>  If you are using kubeadm-dind-cluster, use the command kubectl describe node kube-node-2 | grep InternalIP to determine the value for api_host_name

How come one can be so sure that the nginx pod should be installed on kube-node-2? 
Maybe I am getting it wrong but it sounds like the nginx will be installed on kube-node-2, no wonder, so we only need to grep InternalIP of the kube-node-2.

Actually, last time I saw the nginx was on kube-node-1 but wsk was properly working. I have no idea why it works. Please point out which part of my reasoning is not right.

FYI,
(I am on my ubuntu on macbook)
```
jinc@jinc-MacBookPro:~/github$ kubectl get nodes --show-labels
NAME          STATUS    ROLES     AGE       VERSION   LABELS
kube-master   Ready     master    4m        v1.10.5   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-master,node-role.kubernetes.io/master=
kube-node-1   Ready     <none>    3m        v1.10.5   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-node-1,openwhisk-role=invoker
kube-node-2   Ready     <none>    3m        v1.10.5   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=kube-node-2,openwhisk-role=invoker
```

**Comment by JiniousChoi** [comment]
Now the heath of invokers are properly checked. But I have another issue.
```
jinc@jinc-MacBookPro:~/github$ kubectl get all -n openwhisk -o wide
NAME                              READY     STATUS      RESTARTS   AGE       IP            NODE
pod/apigateway-7b87dd957f-ng29r   1/1       Running     0          6m        10.244.2.5    kube-node-1
pod/controller-0                  1/1       Running     0          6m        10.244.2.4    kube-node-1
pod/couchdb-6b87ccfb78-hvqjv      1/1       Running     0          6m        10.244.3.4    kube-node-2
pod/init-couchdb-gn7b7            0/1       Completed   0          6m        10.244.2.8    kube-node-1
pod/install-catalog-8x7qq         0/1       Error       0          6m        10.244.3.7    kube-node-2
pod/install-catalog-bnj8f         0/1       Error       0          58s       10.244.3.9    kube-node-2
pod/install-catalog-d7c6j         0/1       Error       0          38s       10.244.3.10   kube-node-2
pod/install-catalog-s22cm         0/1       Error       0          1m        10.244.2.10   kube-node-1
pod/install-routemgmt-dhfzr       0/1       Error       0          47s       10.244.2.11   kube-node-1
pod/install-routemgmt-szlf8       0/1       Error       0          6m        10.244.2.7    kube-node-1
pod/install-routemgmt-vh5lp       1/1       Running     0          5s        10.244.2.12   kube-node-1
pod/invoker-r48jz                 1/1       Running     0          6m        10.244.2.3    kube-node-1
pod/invoker-sq8nr                 1/1       Running     0          6m        10.244.3.6    kube-node-2
pod/kafka-0                       1/1       Running     0          6m        10.244.2.9    kube-node-1
pod/nginx-598c75d7d6-6c5sl        1/1       Running     0          6m        10.244.2.6    kube-node-1
pod/redis-5d77674f65-csl6d        1/1       Running     0          6m        10.244.3.5    kube-node-2
pod/zookeeper-0                   1/1       Running     0          6m        10.244.3.8    kube-node-2

NAME                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                     AGE       SELECTOR
service/apigateway   ClusterIP   10.104.72.202    <none>        8080/TCP,9000/TCP                           6m        name=apigateway
service/controller   ClusterIP   10.102.29.83     <none>        8080/TCP                                    6m        name=controller
service/couchdb      ClusterIP   10.107.45.222    <none>        5984/TCP                                    6m        name=couchdb
service/kafka        ClusterIP   10.111.233.124   <none>        9092/TCP                                    6m        name=kafka
service/nginx        NodePort    10.105.80.76     <none>        80:31520/TCP,443:31001/TCP,8443:30304/TCP   6m        name=nginx
service/redis        ClusterIP   10.103.8.64      <none>        6379/TCP                                    6m        name=redis
service/zookeeper    ClusterIP   None             <none>        2181/TCP,2888/TCP,3888/TCP                  6m        name=zookeeper

NAME                     DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE       CONTAINERS   IMAGES                     SELECTOR
daemonset.apps/invoker   2         2         2         2            2           <none>          6m        invoker      openwhisk/invoker:latest   name=invoker

NAME                         DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE       CONTAINERS   IMAGES                        SELECTOR
deployment.apps/apigateway   1         1         1            1           6m        apigateway   openwhisk/apigateway:latest   name=apigateway
deployment.apps/couchdb      1         1         1            1           6m        couchdb      apache/couchdb:2.1            name=couchdb
deployment.apps/nginx        1         1         1            1           6m        nginx        nginx:1.11                    name=nginx
deployment.apps/redis        1         1         1            1           6m        redis        redis:3.2                     name=redis

NAME                                    DESIRED   CURRENT   READY     AGE       CONTAINERS   IMAGES                        SELECTOR
replicaset.apps/apigateway-7b87dd957f   1         1         1         6m        apigateway   openwhisk/apigateway:latest   name=apigateway,pod-template-hash=3643885139
replicaset.apps/couchdb-6b87ccfb78      1         1         1         6m        couchdb      apache/couchdb:2.1            name=couchdb,pod-template-hash=2643779634
replicaset.apps/nginx-598c75d7d6        1         1         1         6m        nginx        nginx:1.11                    name=nginx,pod-template-hash=1547318382
replicaset.apps/redis-5d77674f65        1         1         1         6m        redis        redis:3.2                     name=redis,pod-template-hash=1833230921

NAME                          DESIRED   CURRENT   AGE       CONTAINERS   IMAGES
statefulset.apps/controller   1         1         6m        controller   openwhisk/controller:latest
statefulset.apps/kafka        1         1         6m        kafka        wurstmeister/kafka:0.11.0.1
statefulset.apps/zookeeper    1         1         6m        zookeeper    zookeeper:3.4

NAME                          DESIRED   SUCCESSFUL   AGE       CONTAINERS     IMAGES                            SELECTOR
job.batch/init-couchdb        1         1            6m        init-couchdb   openwhisk/ansible-runner:latest   controller-uid=11e7da9c-e0cb-11e8-a6b0-363c7332d8e7
job.batch/install-catalog     1         0            6m        catalog        openwhisk/script-runner:latest    controller-uid=11e96ab1-e0cb-11e8-a6b0-363c7332d8e7
job.batch/install-routemgmt   1         0            6m        routemgmt      openwhisk/script-runner:latest    controller-uid=11e9ecbf-e0cb-11e8-a6b0-363c7332d8e7
```

## logs for pod/install-catalog-s22cm
```
jinc@jinc-MacBookPro:~/github$ kubectl logs pod/install-catalog-s22cm -n openwhisk
+ '[' -f /task/myTask.sh ']'
+ . /task/myTask.sh
++ export OPENWHISK_HOME=/openwhisk
++ OPENWHISK_HOME=/openwhisk
++ git clone https://github.com/apache/incubator-openwhisk-catalog openwhisk-catalog
Cloning into 'openwhisk-catalog'...
++ pushd openwhisk-catalog/packages
++ ./installCatalog.sh 789c46b1-71f6-4ed5-8c54-816aa4f8c502:abczO3xZCLrMN6v2BKK1dXYFpXlPkccOFqm12CdAsMgRU4VrNZ9lyGVCGuMDGIwP nil:nil /usr/local/bin/wsk
/openwhisk-catalog/packages /
Installing OpenWhisk packages
Installing package installCombinators.sh with pid 24
Installing package installGit.sh with pid 25
Installing package installSlack.sh with pid 26
Installing package installSystem.sh with pid 27
Installing package installWatson.sh with pid 28
Installing package installWeather.sh with pid 29
Installing package installWebSocket.sh with pid 30
Installing Git package.
Creating package github with pid 54
Installing Slack package.
Installing action combinator package.
Creating package combinators with pid 71
Creating package slack with pid 72
Installing WebSocket package.
Installing Weather package.
Creating package websocket with pid 87
Creating package weather with pid 89
Installing whisk.system entities.
Creating package utils with pid 91
Installing Watson package.
Creating package samples with pid 92
Creating package watson-speechToText with pid 93
Creating package watson-textToSpeech with pid 97
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/github?overwrite=true: invalid URL port "nil"
54 finished with status 3
Installing github/webhook with pid 120
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/combinators?overwrite=true: invalid URL port "nil"
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/websocket?overwrite=true: invalid URL port "nil"
87 finished with status 3
71 finished with status 3
Installing websocket/send with pid 121
Installing combinators/eca with pid 122
Installing combinators/forwarder with pid 123
Installing combinators/retry with pid 124
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/slack?overwrite=true: invalid URL port "nil"
Installing combinators/trycatch with pid 125
72 finished with status 3
Installing slack/post with pid 141
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/samples?overwrite=true: invalid URL port "nil"
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/weather?overwrite=true: invalid URL port "nil"
89 finished with status 3
Installing weather/forecast with pid 154
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/utils?overwrite=true: invalid URL port "nil"
91 finished with status 3
92 finished with status 3
Installing utils/echo with pid 155
Installing utils/cat with pid 156
Installing utils/smash with pid 160
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/watson-speechToText?overwrite=true: invalid URL port "nil"
Installing utils/split with pid 161
93 finished with status 3
Installing utils/sort with pid 166
Installing utils/head with pid 167
Installing utils/date with pid 169
Installing utils/namespace with pid 172
Installing utils/hosturl with pid 176
error: Package update failed: Put https://nil:nil/api/v1/namespaces/_/packages/watson-textToSpeech?overwrite=true: invalid URL port "nil"
Installing samples/helloWorld with pid 180
Installing samples/greeting with pid 181
Installing samples/wordCount with pid 185
Installing samples/curl with pid 189
97 finished with status 3
error: Invalid argument(s). Unable to get action 'combinators/trycatch': Get https://nil:nil/api/v1/namespaces/_/actions/combinators/trycatch?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
Installing watson-speechToText/speechToText with pid 209
error: Invalid argument(s). Unable to get action 'combinators/retry': Get https://nil:nil/api/v1/namespaces/_/actions/combinators/retry?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
Installing watson-textToSpeech/textToSpeech with pid 214
error: Invalid argument(s). Unable to get action 'github/webhook': Get https://nil:nil/api/v1/namespaces/_/actions/github/webhook?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
120 finished with status 3
Git package ERRORS = 2
error: Invalid argument(s). Unable to get action 'combinators/eca': Get https://nil:nil/api/v1/namespaces/_/actions/combinators/eca?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
122 finished with status 3
error: Invalid argument(s). Unable to get action 'slack/post': Get https://nil:nil/api/v1/namespaces/_/actions/slack/post?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
141 finished with status 3
Slack package ERRORS = 2
error: Invalid argument(s). Unable to get action 'samples/curl': Get https://nil:nil/api/v1/namespaces/_/actions/samples/curl?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'samples/greeting': Get https://nil:nil/api/v1/namespaces/_/actions/samples/greeting?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/head': Get https://nil:nil/api/v1/namespaces/_/actions/utils/head?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'websocket/send': Get https://nil:nil/api/v1/namespaces/_/actions/websocket/send?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/echo': Get https://nil:nil/api/v1/namespaces/_/actions/utils/echo?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
155 finished with status 3
121 finished with status 3
WebSocket package ERRORS = 2
error: Invalid argument(s). Unable to get action 'samples/wordCount': Get https://nil:nil/api/v1/namespaces/_/actions/samples/wordCount?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/smash': Get https://nil:nil/api/v1/namespaces/_/actions/utils/smash?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/split': Get https://nil:nil/api/v1/namespaces/_/actions/utils/split?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'weather/forecast': Get https://nil:nil/api/v1/namespaces/_/actions/weather/forecast?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'samples/helloWorld': Get https://nil:nil/api/v1/namespaces/_/actions/samples/helloWorld?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
154 finished with status 3
Weather package ERRORS = 2
error: Invalid argument(s). Unable to get action 'utils/cat': Get https://nil:nil/api/v1/namespaces/_/actions/utils/cat?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'combinators/forwarder': Get https://nil:nil/api/v1/namespaces/_/actions/combinators/forwarder?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
156 finished with status 3
160 finished with status 3
123 finished with status 3
124 finished with status 3
125 finished with status 3
combinator package ERRORS = 5
161 finished with status 3
24 finished with status 5
25 finished with status 2
26 finished with status 2
error: Invalid argument(s). Unable to get action 'utils/sort': Get https://nil:nil/api/v1/namespaces/_/actions/utils/sort?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
166 finished with status 3
167 finished with status 3
error: Invalid argument(s). Unable to get action 'utils/namespace': Get https://nil:nil/api/v1/namespaces/_/actions/utils/namespace?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/date': Get https://nil:nil/api/v1/namespaces/_/actions/utils/date?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
169 finished with status 3
172 finished with status 3
error: Invalid argument(s). Unable to get action 'watson-textToSpeech/textToSpeech': Get https://nil:nil/api/v1/namespaces/_/actions/watson-textToSpeech/textToSpeech?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
error: Invalid argument(s). Unable to get action 'utils/hosturl': Get https://nil:nil/api/v1/namespaces/_/actions/utils/hosturl?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
176 finished with status 3
180 finished with status 3
181 finished with status 3
185 finished with status 3
189 finished with status 3
whisk.system entities ERRORS = 15
27 finished with status 15
error: Invalid argument(s). Unable to get action 'watson-speechToText/speechToText': Get https://nil:nil/api/v1/namespaces/_/actions/watson-speechToText/speechToText?code=false: invalid URL port "nil"
Run 'wsk --help' for usage.
209 finished with status 3
214 finished with status 3
Watson package ERRORS = 4
28 finished with status 4
29 finished with status 2
30 finished with status 2
open catalog ERRORS = 7
```

**Comment by JiniousChoi** [comment]
It worked! But only once.
Nwo I am with another issue #336
